tests/sentry/remote_subscriptions/consumers/test_queue_consumer.py

```diff
@@ -0,0 +1,434 @@
+"""
+Tests for the thread-queue-parallel result consumer implementation.
+"""
+
+import threading
+from datetime import datetime
+from typing import Any
+from unittest import mock
+
+from arroyo.backends.kafka import KafkaPayload
+from arroyo.types import BrokerValue, FilteredPayload, Message, Partition, Topic
+
+from sentry.remote_subscriptions.consumers.queue_consumer import (
+    FixedQueuePool,
+    OffsetTracker,
+    SimpleQueueProcessingStrategy,
+    WorkItem,
+)
+from sentry.testutils.cases import TestCase
+
+
+class TestOffsetTracker(TestCase):
+    def setUp(self):
+        self.tracker = OffsetTracker()
+        self.partition1 = Partition(Topic("test"), 0)
+        self.partition2 = Partition(Topic("test"), 1)
+
+    def test_simple_tracking(self):
+        """Test basic offset tracking and committing."""
+        self.tracker.add_offset(self.partition1, 100)
+        self.tracker.add_offset(self.partition1, 101)
+        self.tracker.add_offset(self.partition1, 102)
+
+        committable = self.tracker.get_committable_offsets()
+        assert committable == {}
+
+        self.tracker.complete_offset(self.partition1, 100)
+        committable = self.tracker.get_committable_offsets()
+        assert committable == {self.partition1: 100}
+
+        self.tracker.mark_committed(self.partition1, 100)
+        self.tracker.complete_offset(self.partition1, 102)
+        committable = self.tracker.get_committable_offsets()
+        assert committable == {}
+
+        self.tracker.complete_offset(self.partition1, 101)
+        committable = self.tracker.get_committable_offsets()
+        assert committable == {self.partition1: 102}
+
+    def test_multiple_partitions(self):
+        """Test tracking across multiple partitions."""
+        self.tracker.add_offset(self.partition1, 100)
+        self.tracker.add_offset(self.partition1, 101)
+        self.tracker.add_offset(self.partition2, 200)
+        self.tracker.add_offset(self.partition2, 201)
+
+        self.tracker.complete_offset(self.partition1, 100)
+        self.tracker.complete_offset(self.partition2, 200)
+        self.tracker.complete_offset(self.partition2, 201)
+
+        committable = self.tracker.get_committable_offsets()
+        assert committable == {self.partition1: 100, self.partition2: 201}
+
+    def test_handles_gaps(self):
+        """Test that we never commit past gaps."""
+        self.tracker.add_offset(self.partition1, 100)
+        self.tracker.add_offset(self.partition1, 102)
+        self.tracker.add_offset(self.partition1, 103)
+
+        self.tracker.complete_offset(self.partition1, 100)
+        self.tracker.complete_offset(self.partition1, 102)
+        self.tracker.complete_offset(self.partition1, 103)
+
+        committable = self.tracker.get_committable_offsets()
+        assert committable == {self.partition1: 100}
```

Comment:
if there's a gap doesn't this mean the next time it commits it would commit 102 then?

when would we end up with gaps like this?